src/Lean/Meta/Match/MatchEqs.lean

```diff
@@ -88,13 +102,11 @@ where
       <|>
       (splitSparseCasesOn mvarId)
       <|>
-      (do let mvarId' ← simpIfTarget mvarId (useDecide := true) (useNewSemantics := true)
```

Comment:
It seems that this call is not only effective after an explicit split, but actually picks up equations from the context.

So maybe better this way.

Or the new function above should scan the local context, still better than running simp in general.